include/psa/crypto_values.h

```diff
@@ -1678,6 +1691,78 @@
+ * Implementations of the present specification use Schnorr NIZKP and this does
+ * not need to be configured in the cipher suites.
+ *
+ * J-PAKE can be used with any secure cryptographic hash function, the choice
+ * of hash must be supplied to the psa_pake_cipher_suite() as the second
+ * parameter (\p hash).
+ *
+ * All the remaining parameters passed to psa_pake_cipher_suite() when creating
+ * the cipher suite must be 0.
+ *
+ * The key exchange flow for JPAKE is as follows:
+ * -# To get the first round data that needs to be sent to the peer, call
+ *      // Get g1
+ *      psa_pake_output(operation, #PSA_PAKE_DATA_KEY_SHARE, ...);
+ *      // Get the ZKP public key for x1
+ *      psa_pake_output(operation, #PSA_PAKE_DATA_ZK_PUBLIC, ...);
+ *      // Get the ZKP proof for x1
+ *      psa_pake_output(operation, #PSA_PAKE_DATA_ZK_PROOF, ...);
+ *      // Get g2
+ *      psa_pake_output(operation, #PSA_PAKE_DATA_KEY_SHARE_2, ...);
+ *      // Get the ZKP public key for x2
+ *      psa_pake_output(operation, #PSA_PAKE_DATA_ZK_PUBLIC_2, ...);
+ *      // Get the ZKP proof for x2
+ *      psa_pake_output(operation, #PSA_PAKE_DATA_ZK_PROOF_2, ...);
+ * -# To provide the first round data received from the peer to the operation,
+ *    call
+ *      // Set g3
+ *      psa_pake_input(operation, #PSA_PAKE_DATA_KEY_SHARE, ...);
+ *      // Set the ZKP public key for x3
+ *      psa_pake_input(operation, #PSA_PAKE_DATA_ZK_PUBLIC, ...);
+ *      // Set the ZKP proof for x3
+ *      psa_pake_input(operation, #PSA_PAKE_DATA_ZK_PROOF, ...);
+ *      // Set g4
+ *      psa_pake_input(operation, #PSA_PAKE_DATA_KEY_SHARE_2, ...);
+ *      // Set the ZKP public key for x4
+ *      psa_pake_input(operation, #PSA_PAKE_DATA_ZK_PUBLIC_2, ...);
+ *      // Set the ZKP proof for x4
+ *      psa_pake_input(operation, #PSA_PAKE_DATA_ZK_PROOF_2, ...);
+ * -# To get the second round data that needs to be sent to the peer, call
+ *      // Get A
+ *      psa_pake_output(operation, #PSA_PAKE_DATA_KEY_SHARE_3, ...);
+ *      // Get ZKP public key for x2*s
+ *      psa_pake_output(operation, #PSA_PAKE_DATA_ZK_PUBLIC_3, ...);
+ *      // Get ZKP proof for x2*s
+ *      psa_pake_output(operation, #PSA_PAKE_DATA_ZK_PROOF_3, ...);
+ * -# To provide the second round data received from the peer to the operation,
+ *    call
+ *      // Set B
+ *      psa_pake_input(operation, #PSA_PAKE_DATA_KEY_SHARE_3, ...);
+ *      // Set ZKP public key for x4*s
+ *      psa_pake_input(operation, #PSA_PAKE_DATA_ZK_PUBLIC_3, ...);
+ *      // Set ZKP proof for x4*s
+ *      psa_pake_input(operation, #PSA_PAKE_DATA_ZK_PROOF_3, ...);
+ * -# To access the shared secret call
+ *      // Get Ka=Kb=K
+ *      psa_pake_get_implicit_key()
+ *
+ * For more information consult the documentation of the individual
+ * PSA_PAKE_DATA_XXX constants.
+ *
+ * J-PAKE is standardised for example in RFC 8236.
+ */
+#define PSA_ALG_PAKE_JPAKE                   ((psa_algorithm_t)0x0a000001)
```

Comment:
In the algorithm encoding used by PSA Crypto, bits[7:0] are _always_ used to identify a hash algorithm.

I suspect that the `0x01` value used in this encoding is to identify the type of PAKE algorithm? In which case, unless there is a reason to use bits[15:8] for something else, it would be more consistent to use bits[15:8] for this information, and leave bits[7:0] as zero.

Reply:
Is the use of a secure hash algorithm common within PAKE constructions? - If so, is it worth encoding the hash used with J-PAKE into the algorithm identifier?

An alternative approach to the use of the algorithm identifier encoding space would be to do something similar to the key-agreement + KDF, and allow a composite PAKE + KDF algorithm identifier (as well as the raw PAKE)?

Reply:
I have updated the value. I didn't encode the hash in the last byte, I don't think it would worth it, because it is never enough to fully characterise a PAKE.